bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/EvoCommunicator.java

```diff
@@ -77,15 +77,16 @@ protected void receiveRamResponse(IResponse response) throws ParadoxException {
         int ramBlockNumber = request.getRamBlockNumber();
 
         byte[] parsedResult = parsePacket(response);
-        if (parsedResult != null) {
+        if (parsedResult != null && parsedResult.length == RAM_BLOCK_SIZE) {
             memoryMap.updateElement(ramBlockNumber, parsedResult);
             logger.trace("Result for ramBlock={} is [{}]", ramBlockNumber, parsedResult);
         } else {
             logger.debug("Wrong parsed result. Probably wrong data received in response");
+            return;
         }
 
         // Trigger listeners update when last memory page update is received
-        if (ramBlockNumber == panelType.getRamPagesNumber() && listeners != null && !listeners.isEmpty()) {
+        if (ramBlockNumber == panelType.getRamPagesNumber()) {

```

Comment:
Isn't that very risky? Why are you sure that listeners is not null? Otherweise we'll get an NPE https://github.com/openhab/openhab2-addons/blob/34d0f4760e497ba0c092a31381bc440b55f2f885/bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/GenericCommunicator.java#L146

Reply:
Forget that. I did not look at the changed implementation in the `GeneriCommunicator`. That would be easier if we had null annotations....

Reply:
Yep. I moved that inside the method. I think it's better this way...